scripts/builtin/dist.dml

```diff
@@ -32,7 +32,8 @@
 # -----------------------------------------------------------------------------------------------
 
 m_dist = function(Matrix[Double] X) return (Matrix[Double] Y) {
-  G = X %*% t(X);
-  Y = sqrt(-2 * G + outer(diag(G), t(diag(G)), "+"));
+  n = nrow(X)
+  s = rowSums(X^2)
+  Y = sqrt(-2 * X %*% t(X) + s + t(s))
```

Comment:
if you can limit it to not include this change in both PRs it would be great.
Alternatively you can move this specific dist change to a separate PR.

Reply:
We already did a PR for the dist function, so is it better to revert the changes of dist() for the current PR?